consensus/ethash/consensus.go

```diff
@@ -264,14 +264,25 @@ func (ethash *Ethash) verifyHeader(chain consensus.ChainHeaderReader, header, pa
 	if expected.Cmp(header.Difficulty) != 0 {
 		return fmt.Errorf("invalid difficulty: have %v, want %v", header.Difficulty, expected)
 	}
+
+	// Verify that the "GasLimit" header field is valid
 	// Verify that the gas limit is <= 2^63-1
-	cap := uint64(0x7fffffffffffffff)
-	if header.GasLimit > cap {
-		return fmt.Errorf("invalid gasLimit: have %v, max %v", header.GasLimit, cap)
+	max := uint64(0x7fffffffffffffff)
+	if header.GasLimit > max {
+		return fmt.Errorf("invalid gasLimit: have %v, max %v", header.GasLimit, max)
+	}
+
+	// Verify that total gas usage is below the gas limit
+	// Before EIP1559 activation, the "gasLimit" field in the header directly represents the gas limit
+	// After EIP1559 activation, the "gasLimit" field in the header represents the total gas target
+	// After EIP1559 activation, the *total* gas limit is equal to total gas target * slack coefficient
+	// The individual limits on legacy and EIP1559 gas usage during the transition period are enforced in the core/block_validator.go
+	var slackCoefficient uint64 = 1
```

Comment:
I think the default slack coefficient should be 2. And maybe we should consider renaming it to `ELASTICITY_MULTIPLIER` to match the specification